AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java

```diff
@@ -19,31 +20,37 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         final AWSMobileClient auth = AWSMobileClient.getInstance();
+        final AppCompatActivity thisActivity = this;
 
-        if (auth.isSignedIn()) {
-            ActivityUtils.startActivity(this, SimpleNavActivity.class);
-        } else {
-            auth.showSignIn(this,
-                    SignInUIOptions.builder()
-                            .nextActivity(SimpleNavActivity.class)
-                            .build(),
-                    new Callback<UserStateDetails>() {
-                        @Override
-                        public void onResult(UserStateDetails result) {
-                            Log.d(TAG, "onResult: User signed-in " + result.getUserState());
-                        }
+        AsyncTask.execute(new Runnable() {
+            @Override
+            public void run() {
+                if (auth.isSignedIn()) {
```

Comment:
Is it possible to just call `isSignedIn()` from a background thread or AsyncTask and wait until the call returns and then proceed to start activity or showSignIn based on the response?

Reply:
I would rather expect AWS mobile client will do this Async task and return response inside the isSignedIn().

Reply:
auth.getTokens(new Callback<Tokens>() {
                        @Override
                        public void onResult(Tokens result) {
                            ActivityUtils.startActivity(thisActivity, SimpleNavActivity.class);
                        }

                        @Override
                        public void onError(Exception e) {

                        }
                    });